### PR TITLE
wait function: changed to have an option to use it with timeout

### PR DESCRIPTION
**Pull request by larisau**

### CODE REVIEW

**Comment by larisau** [comment]
You're right, args will never work here.
I can just remove it, because it's not used in any place) or change as suggested.

**Comment by larisau** [comment]
removed args to be compatible with an existing code using kwargs

File: sdcm/wait.py [hunk 1]
```diff
@@ -30,14 +65,13 @@ def wait_for(func, step=1, text=None, *args, **kwargs):
     :param func: Function to evaluate.
     :param step: Amount of time to sleep before another try.
     :param text: Text to log, for debugging purposes.
-    :param args: Positional arguments to func
     :param kwargs: Keyword arguments to func
```
Comment (amoskong) [inline sdcm/wait.py]: :param kwargs: Keyword arguments to func, contains both converted non-keyword arguments and original keyword arguments. Example: foo(arg1, arg2=None) -> forever_wait_for(foo, arg1='v1', arg2='v2')

File: sdcm/wait.py [hunk 2]
```diff
@@ -16,11 +16,46 @@
...
+    Wrapper function to wait with timeout option.
+    If timeout received, avocado 'wait_for' method will be used.
+    Otherwise the below function will be called.
+
+    :param func: Function to evaluate.
+    :param step: Time to sleep between attempts in seconds
+    :param text: Text to print while waiting, for debug purposes
+    :param timeout: Timeout in seconds
+    :param throw_exc: Raise exception if timeout expired, but func result is not True
+    :param kwargs: Keyword arguments to func
```
Comment (amoskong) [inline sdcm/wait.py]: :param kwargs: Keyword arguments to func, contains both converted non-keyword arguments and original keyword arguments. Example: foo(arg1, arg2=None) -> wait_for(foo, arg1='v1', arg2='v2')

Comment (larisau) [reply sdcm/wait.py]: I removed an optional positional args, so you can only pass keyword args.
This is an example from the existing code(see cluster.py):
wait.wait_for(func=self.cfstat_reached_threshold, step=10,
                      **text=text, key=key, threshold=threshold**)

File: sdcm/wait.py (hunk 1 above)
Comment (larisau) [reply sdcm/wait.py]: This code will work

Comment (amoskong) [reply sdcm/wait.py]: Current code works. If people who uses wait_for() in future has concern about passing non-keyword arguments of func, they can find helper by above description.

**Pull request merged**